implementations/micrometer-registry-otlp/src/test/java/io/micrometer/registry/otlp/OtlpDeltaMeterRegistryTest.java

```diff
@@ -311,4 +312,69 @@ void testMetricsStartAndEndTime() {
         assertThat(getDataPoint.apply(counter).getTimeUnixNano()).isEqualTo(120000000000L);
     }
 
+    @Test
+    void scheduledRollOver() {
+        Counter counter = Counter.builder(METER_NAME)
+            .description(METER_DESCRIPTION)
+            .tags(Tags.of(meterTag))
+            .register(registry);
+
+        AtomicLong functionCount = new AtomicLong(15);
+        FunctionCounter functionCounter = FunctionCounter.builder("counter.function", functionCount, AtomicLong::get)
+            .register(registry);
+        FunctionTimer functionTimer = FunctionTimer
+            .builder("timer.function", functionCount, AtomicLong::get, AtomicLong::get, MILLISECONDS)
+            .register(registry);
+
+        // before rollover
+        assertSum(writeToMetric(counter), 0, TimeUnit.MINUTES.toNanos(1), 0);
+        assertThat(functionCounter.count()).isZero();
+        assertThat(functionTimer.count()).isZero();
+        assertThat(functionTimer.totalTime(MILLISECONDS)).isZero();
+
+        this.stepOverNStep(1);
```

Comment:
The implementation looks ok but I don't think we are testing here what we should.
Since the test is initialized with adding a step to the clock, this is already the second step. This is even weirder if you add `counter.increment();` right after the `Counter` creation (which I think we should), in that case `counter.count()` should be 0 but will be 1 since we are already passed the step boundary.

Reply:
Thanks for catching that. I missed updating the meter in the same step. Fixed this.